scvelo/tools/_em_model.py

```diff
@@ -239,42 +249,30 @@ def fit(
 
         for ix, dm in zip(idx, dms):
             T[:, ix], Tau[:, ix], Tau_[:, ix] = dm.t, dm.tau, dm.tau_
-            alpha[ix], beta[ix], gamma[ix], t_[ix], scaling[ix] = dm.pars[:, -1]
-            u0[ix], s0[ix], pval[ix] = dm.u0, dm.s0, dm.pval_steady
-            steady_u[ix], steady_s[ix] = dm.steady_u, dm.steady_s
-            beta[ix] /= scaling[ix]
-            steady_u[ix] *= scaling[ix]
-
-            std_u[ix], std_s[ix] = dm.std_u, dm.std_s
-            likelihood[ix], varx[ix] = dm.likelihood, dm.varx
+            (
+                sd.alpha[ix],
+                sd.beta[ix],
+                sd.gamma[ix],
+                sd.t_[ix],
+                sd.scaling[ix],
+            ) = dm.pars[:, -1]
+            sd.u0[ix], sd.s0[ix], sd.pval_steady[ix] = dm.u0, dm.s0, dm.pval_steady
+            sd.steady_u[ix], sd.steady_s[ix] = dm.steady_u, dm.steady_s
+            sd.beta[ix] /= sd.scaling[ix]
+            sd.steady_u[ix] *= sd.scaling[ix]
+
+            sd.std_u[ix], sd.std_s[ix] = dm.std_u, dm.std_s
+            sd.likelihood[ix], sd.variance[ix] = dm.likelihood, dm.varx
             L.append(dm.loss)
 
-        _pars = [
-            alpha,
-            beta,
-            gamma,
-            t_,
-            scaling,
-            std_u,
-            std_s,
-            likelihood,
-            u0,
-            s0,
-            pval,
-            steady_u,
-            steady_s,
-            varx,
-        ]
-
-        adata = self._adata.copy() if copy else self._adata
+        adata = self.export_results_adata(copy=copy, add_key=self._fit_key)
 
         adata.uns["recover_dynamics"] = {
             "fit_connected_states": fit_connected_states,
             "fit_basal_transcription": self._fit_basal_transcription,
             "use_raw": use_raw,
         }
 
-        _write_pars(adata, _pars, add_key=self._fit_key)
```

Comment:
Similar to the implemented classes, can you please revert these changes and open a separate PR for them? Or am I missing something, and all of it is needed when switching to using `EMParams`?